openjpa-jdbc/src/main/java/org/apache/openjpa/jdbc/schema/Table.java

```diff
@@ -312,11 +315,20 @@ public String getResourceName() {
         return _rels;
     }
 
+    /**
+     * Return the list of column names, used only for informative (error) messages.
+     * @return
+     */
     public String[] getColumnNames() {
         if (_colMap == null) {
             return new String[0];
         }
-        DBIdentifier[] sNames = _colMap.keySet().toArray(new DBIdentifier[_colMap.size()]);
```

Comment:
`_colMap.keySet()` contains an internal representation of column names (all uppercased)
this method is used only for error messages and IMHO it is better to have a consistent handling of this internal data structure

Reply:
_colMap
                .values()
                .stream()
                .map(Column::getIdentifier)
                .toArray(DBIdentifier[]::new);

Reply:
@rmannibucau 
I can't get your suggestion
If I change this to:
```
_colMap
.values()
.stream()
.map(Column::getIdentifier)
.toArray(DBIdentifier[]::new);
```
I get a DBIdentifier[] and not a String[]

Reply:
done

Reply:
Proposal was just to not to toList().toArray which allocates twice for no real reason the same structure. Probably missed a .map in my snippet but the double allocation should be avoided IMHO.

Reply:
yes I understood and I have updated the code, thanks

Reply:
Hmm, maybe diff is not up to date but you still allocate 2 arrays, first one is a DBIdentifier array instead of an ArrayList but it is ~the same size (I'm thinking to the case we have > 200 columns in the db and it happens) so I would just do:

_colMap
.values()
.stream()
.map(Column::getIdentifier)
.map(DBIdentifier::getName)
.toArray(String[]::new);

Reply:
Still a double alloc no?